src/xdg.rs

```diff
@@ -262,6 +262,63 @@ impl IconLoader {
         Self { icons }
     }
 
+    fn premultiply_generic(data: &mut [u8]) {
+        // TODO: change to array_chunks_mut() once that is stabilised.
+        for chunk in data.chunks_exact_mut(4) {
+            if let [r, g, b, a] = chunk {
+                let r = *r as u16 * *a as u16 + 127;
+                let g = *g as u16 * *a as u16 + 127;
+                let b = *b as u16 * *a as u16 + 127;
+                chunk[0] = ((r + (r >> 8) + 1) >> 8) as u8;
+                chunk[1] = ((g + (g >> 8) + 1) >> 8) as u8;
+                chunk[2] = ((b + (b >> 8) + 1) >> 8) as u8;
+            }
+        }
+    }
+
+    #[cfg(all(target_arch = "aarch64", target_feature = "neon"))]
+    unsafe fn premultiply(data: &mut [u8]) {
```

Comment:
This function is safe, so I think we shouldn't mark it as unsafe. Having the entire body be unsafe might screw with formatting a bit, but I think it might be a little more clear? Having a separate wrapper function might also be an option.